test/benchmarks/transpiler_hls_plugins.py

```diff
@@ -0,0 +1,227 @@
+        q = QuantumRegister(1, "q")
+        qc = QuantumCircuit(q)
+        qc.append(VGate(), [q[0]], [])
+        qc.append(VGate(), [q[0]], [])
+        self.definition = qc
+
+class HLSPluginsSuite:
+
+    # randomized benchmarking
+    def layer_fidelity_clifford_circuit(self, num_qubits, depth, seed=None):
+        qc = QuantumCircuit(num_qubits)
+        for i in range(0, depth):
+            for _ in range(depth):
+                cliff = random_clifford(2, seed=seed)
+                qc.append(cliff, [2 * i, 2 * i + 1])
+        return qc
+
+    # single large clifford
+    def random_clifford_circuit(self, num_qubits, num_gates, gates="all", seed=None):
+        """Generate a pseudo random Clifford circuit."""
+
+        qubits_1_gates = ["i", "x", "y", "z", "h", "s", "sdg", "sx", "sxdg", "v", "w"]
+        qubits_2_gates = ["cx", "cz", "cy", "swap", "iswap", "ecr", "dcx"]
+        if gates == "all":
+            if num_qubits == 1:
+                gates = qubits_1_gates
+            else:
+                gates = qubits_1_gates + qubits_2_gates
+
+        instructions = {
+            "i": (IGate(), 1),
+            "x": (XGate(), 1),
+            "y": (YGate(), 1),
+            "z": (ZGate(), 1),
+            "h": (HGate(), 1),
+            "s": (SGate(), 1),
+            "sdg": (SdgGate(), 1),
+            "sx": (SXGate(), 1),
+            "sxdg": (SXdgGate(), 1),
+            "v": (VGate(), 1),
+            "w": (WGate(), 1),
+            "cx": (CXGate(), 2),
+            "cy": (CYGate(), 2),
+            "cz": (CZGate(), 2),
+            "swap": (SwapGate(), 2),
+            "iswap": (iSwapGate(), 2),
+            "ecr": (ECRGate(), 2),
+            "dcx": (DCXGate(), 2),
+        }
+
+        if isinstance(seed, np.random.Generator):
+            rng = seed
+        else:
+            rng = np.random.default_rng(seed)
+
+        samples = rng.choice(gates, num_gates)
+
+        circ = QuantumCircuit(num_qubits)
+
+        for name in samples:
+            gate, nqargs = instructions[name]
+            qargs = rng.choice(range(num_qubits), nqargs, replace=False).tolist()
+            circ.append(gate, qargs)
+
+        return circ
+
+    def setUp(self):
+        self.num_qubits = 150
+        self.rng = np.random.default_rng(1234)
+
+        # generate num.samples
+        mat = random_invertible_binary_matrix(self.num_qubits)
+        self.qc_linear = QuantumCircuit(self.num_qubits)
+        self.qc_linear.append(LinearFunction(mat), list(range(self.num_qubits)), copy=False)
+
+        self.num_samples = 20
+
+        # separate transpile from build
+        # track: depth, num. 2q gates, 2q depth
+
+    # Clifford Synthesis Plugins
+    def time_clifford_ag_large_circuit(self):
+        """Test A&G synthesis for set of {num_qubits}-qubit Cliffords"""
+        hls_config = HLSConfig(clifford=["ag"])
+        pm = PassManager([HighLevelSynthesis(hls_config=hls_config)])
+
+        for _ in range(self.num_samples):
```

Comment:
We don't need to do this explicitly the asv harness handles taking multiple samples to get the statistical sampling. We probably want to move the random clifford construction to setup too though to avoid randomization impacting the measurements.

Reply:
Should we nonetheless consider _multiple_ different random linear functions, random cliffords, etc. to avoid overtuning synthesis algorithms on specific circuits?